lib/internal/fs.js

```diff
@@ -14,15 +16,189 @@ const {
   O_RDWR,
   O_SYNC,
   O_TRUNC,
-  O_WRONLY
+  O_WRONLY,
+  S_IFBLK,
+  S_IFCHR,
+  S_IFDIR,
+  S_IFIFO,
+  S_IFLNK,
+  S_IFMT,
+  S_IFREG,
+  S_IFSOCK,
+  UV_FS_SYMLINK_DIR,
+  UV_FS_SYMLINK_JUNCTION
 } = process.binding('constants').fs;
+const { statValues } = process.binding('fs');
+
+const isWindows = process.platform === 'win32';
 
 function assertEncoding(encoding) {
   if (encoding && !Buffer.isEncoding(encoding)) {
     throw new errors.TypeError('ERR_INVALID_OPT_VALUE_ENCODING', encoding);
   }
 }
 
+function copyObject(source) {
```

Comment:
I know this PR just moves this - but do you know the rationale for reimplementing this over?

Reply:
I don't

Reply:
Then I guess I'll just follow up on it after this lands